tests/core/test_signal.py

```diff
@@ -208,6 +209,32 @@ async def check_set_and_wait():
     assert await signal.get_value() == 1
 
 
+async def test_set_and_wait_for_value_different_set_and_read():
+    set_signal = epics_signal_rw(int, "pva://set", name="set-signal")
+    read_signal = epics_signal_r(str, "pva://read", name="read-signal")
+    await set_signal.connect(mock=True)
+    await read_signal.connect(mock=True)
+
+    async def wait_and_set_read():
+        await asyncio.sleep(0.1)
+        set_mock_value(read_signal, "test")
+        await asyncio.sleep(0.01)
+
+    async def check_set_and_wait():
+        st = await set_and_wait_for_other_value(
+            set_signal, 1, read_signal, "test", timeout=100
+        )
+        await st
+        await asyncio.sleep(0.01)
+
+    assert (
+        0.1
+        < await time_taken_by(asyncio.gather(wait_and_set_read(), check_set_and_wait()))
+        < 0.15
```

Comment:
0.05s is not long for CI, concerned this test might be flaky in the long run...

Is it possible to use some `asyncio.Events` instead of sleeps to ensure the ordering between the two tasks?